lib/puppet/pops/issues.rb

```diff
@@ -152,680 +154,681 @@ def self.hard_issue(issue_code, *args, &block)
   end
 
   BAD_STRING_SLICE_KEY_TYPE = issue :BAD_STRING_SLICE_KEY_TYPE, :left_value, :actual_type do
-    actual_type_text = "#{label.article(actual_type)} #{actual_type}"
-    "A substring operation does not accept #{actual_type_text} as a character index. Expected an Integer"
+    actual_type_text = _("#{label.article(actual_type)} #{actual_type}")
+    _("A substring operation does not accept %{actual_type_text} as a character index. Expected an Integer") % { actual_type_text: actual_type_text }
   end
 
   BAD_NOT_UNDEF_SLICE_TYPE = issue :BAD_NOT_UNDEF_SLICE_TYPE, :base_type, :actual do
-    "#{base_type}[] argument must be a Type or a String. Got #{actual}"
+    _("%{expression}[] argument must be a Type or a String. Got %{actual}") % { expression: base_type, actual: actual }
   end
 
   BAD_TYPE_SLICE_TYPE = issue :BAD_TYPE_SLICE_TYPE, :base_type, :actual do
-    "#{base_type}[] arguments must be types. Got #{actual}"
+    _("%{base_type}[] arguments must be types. Got %{actual}") % { base_type: base_type, actual: actual }
   end
 
   BAD_TYPE_SLICE_ARITY = issue :BAD_TYPE_SLICE_ARITY, :base_type, :min, :max, :actual do
     base_type_label = base_type.is_a?(String) ? base_type : label.a_an_uc(base_type)
     if max == -1 || max == Float::INFINITY
-      "#{base_type_label}[] accepts #{min} or more arguments. Got #{actual}"
+      _("%{base_type_label}[] accepts %{min} or more arguments. Got %{actual}") % { base_type_label: base_type_label, min: min, actual: actual }
     elsif max && max != min
-      "#{base_type_label}[] accepts #{min} to #{max} arguments. Got #{actual}"
+      _("%{base_type_label}[] accepts %{min} to %{max} arguments. Got %{actual}") % { base_type_label: base_type_label, min: min, max: max, actual: actual }
     else
-      "#{base_type_label}[] accepts #{min} #{label.plural_s(min, 'argument')}. Got #{actual}"
+      _("%{base_type_label}[] accepts %{min} %{label}. Got %{actual}") % { base_type_label: base_type_label, min: min, label: label.plural_s(min, _('argument')), actual: actual }
     end
   end
 
   BAD_TYPE_SPECIALIZATION = hard_issue :BAD_TYPE_SPECIALIZATION, :type, :message do
-    "Error creating type specialization of #{label.a_an(type)}, #{message}"
+    _("Error creating type specialization of %{base_type}, %{message}") % { base_type: label.a_an(type), message: message }
   end
 
   ILLEGAL_TYPE_SPECIALIZATION = issue :ILLEGAL_TYPE_SPECIALIZATION, :kind do
-    "Cannot specialize an already specialized #{kind} type"
+    _("Cannot specialize an already specialized %{kind} type") % { kind: kind }
   end
 
   ILLEGAL_RESOURCE_SPECIALIZATION = issue :ILLEGAL_RESOURCE_SPECIALIZATION, :actual do
-    "First argument to Resource[] must be a resource type or a String. Got #{actual}."
+    _("First argument to Resource[] must be a resource type or a String. Got %{actual}.") % { actual: actual }
```

Comment:
@Magisus @MosesMendoza I'm unsure here, but should we leave Resource[ ] out of translatation?